core/src/test/java/org/apache/calcite/test/TypeCoercionConverterTest.java

```diff
@@ -133,4 +162,24 @@
   private void checkPlanEquals(String sql) {
     tester.assertConvertsTo(sql, "${plan}");
   }
+
+  private void checkValidateSqlEquals(String sql) {
+    Objects.requireNonNull(sql, "sql");
+    final SqlNode sqlQuery;
+    try {
+      sqlQuery = tester.parseQuery(sql);
+    } catch (RuntimeException | Error e) {
+      throw e;
+    } catch (Exception e) {
+      throw TestUtil.rethrow(e);
+    }
+    final RelDataTypeFactory typeFactory = tester.getValidator().getTypeFactory();
+    final Prepare.CatalogReader catalogReader =
+        tester.createCatalogReader(typeFactory);
+    final SqlValidator validator =
+        tester.createValidator(catalogReader, typeFactory);
+
+    final String actual = validator.validate(sqlQuery).toString();
```

Comment:
Somehow we need some refactoring for this testing infrustructure, for e.g, a `Sql` util, we can do that in another PR.

Reply:
I'm interested in it. `checkValidateSqlEquals` function can be abstracted. 
Do you think it's more appropriate to put it in the parent class `SqlToRelTestBase` or create an interface `SqlToValidateTestBase`?

Reply:
@JiajunBernoulli , have you seen https://lists.apache.org/thread/3l9y4fv31kthmpn9pbzqdmfp80yrjphf by @julianhyde ?

Reply:
I didn't know before. I will continue to pay attention to it